server/apps/main/views.py

```diff
@@ -150,7 +152,28 @@ def share_experience(request, uuid=False):
 
                 # redirect to a new URL:
                 return redirect("main:confirm_page")
-
+            # If form is invalid raise errors back to user
+            else:             
+                for field in form:
+                    for error in field.errors:
+                        if field != '__all__':
+                            messages.add_message(request, messages.WARNING, "{}: {}".format(field.label, error))
+                if uuid: # check if editing existing exp
+                    moderation_status = PublicExperience.objects.get(experience_id=uuid).moderation_status
+                    title = "Edit experience"
+                else: # or creating new one
+                    moderation_status = "not reviewed"
+                    title = "Share experience"
+                return render(
+                    request,
+                    "main/share_experiences.html",
+                    {
+                        "form": form,
+                        "uuid": uuid,
+                        "title": title,
+                        "moderation_status": moderation_status,
```

Comment:
David pointed me to there not being a `viewable` variable here.

However, I think the share_experience.html template also doesn't make of the `viewable` variable, although it is returned later in this function (line 201). Should we delete it later on or add it here?

Reply:
I noticed that as well! I think this is a result of how the form later on (in the `else` part for non-POSTs) is being initialized by being a merge from the `PublicExperience` object and the OH file. 

So I wasn't 100% sure if we could remove it there as well - but we probably could, but maybe worth not putting this into the PR here but rather clean up as its own PR? :)

Reply:
sounds good!